src/ImageSharp/ColorProfiles/ColorProfileConverterExtensionsIcc.cs

```diff
@@ -33,64 +34,103 @@ internal static TTo ConvertUsingIccProfile<TFrom, TTo>(this ColorProfileConverte
             MemoryAllocator = converter.Options.MemoryAllocator,
 
             // TODO: Double check this but I think these are normalized values.
-            SourceWhitePoint = CieXyz.FromScaledVector4(new(converter.Options.SourceIccProfile.Header.PcsIlluminant, 1F)),
-            TargetWhitePoint = CieXyz.FromScaledVector4(new(converter.Options.TargetIccProfile.Header.PcsIlluminant, 1F)),
+            SourceWhitePoint = new CieXyz(converter.Options.SourceIccProfile.Header.PcsIlluminant),
+            TargetWhitePoint = new CieXyz(converter.Options.TargetIccProfile.Header.PcsIlluminant),
         });
 
         IccDataToPcsConverter sourceConverter = new(converter.Options.SourceIccProfile);
         IccPcsToDataConverter targetConverter = new(converter.Options.TargetIccProfile);
-        IccColorSpaceType sourcePcsType = converter.Options.SourceIccProfile.Header.ProfileConnectionSpace;
-        IccColorSpaceType targetPcsType = converter.Options.TargetIccProfile.Header.ProfileConnectionSpace;
-        IccVersion sourceVersion = converter.Options.SourceIccProfile.Header.Version;
-        IccVersion targetVersion = converter.Options.TargetIccProfile.Header.Version;
-
-        Vector4 pcs = sourceConverter.Calculate(source.ToScaledVector4());
-
-        // Profile connecting spaces can only be Lab, XYZ.
-        if (sourcePcsType is IccColorSpaceType.CieLab && targetPcsType is IccColorSpaceType.CieXyz)
-        {
-            // Convert from Lab to XYZ.
-            CieLab lab = CieLab.FromScaledVector4(pcs);
-            CieXyz xyz = pcsConverter.Convert<CieLab, CieXyz>(in lab);
-            pcs = xyz.ToScaledVector4();
-        }
-        else if (sourcePcsType is IccColorSpaceType.CieXyz && targetPcsType is IccColorSpaceType.CieLab)
+        IccProfileHeader sourceHeader = converter.Options.SourceIccProfile.Header;
+        IccProfileHeader targetHeader = converter.Options.TargetIccProfile.Header;
+        IccColorSpaceType sourcePcsType = sourceHeader.ProfileConnectionSpace;
+        IccColorSpaceType targetPcsType = targetHeader.ProfileConnectionSpace;
+        IccRenderingIntent sourceIntent = sourceHeader.RenderingIntent;
+        IccRenderingIntent targetIntent = targetHeader.RenderingIntent;
+        IccVersion sourceVersion = sourceHeader.Version;
+        IccVersion targetVersion = targetHeader.Version;
+
+        // all conversions are funnelled through XYZ in case PCS adjustments need to be made
+        CieXyz xyz;
+
+        Vector4 sourcePcs = sourceConverter.Calculate(source.ToScaledVector4());
+        switch (sourcePcsType)
         {
-            // Convert from XYZ to Lab.
-            CieXyz xyz = CieXyz.FromScaledVector4(pcs);
-            CieLab lab = pcsConverter.Convert<CieXyz, CieLab>(in xyz);
-            pcs = lab.ToScaledVector4();
+            case IccColorSpaceType.CieLab:
+                if (sourceConverter.Is16BitLutEntry)
```

Comment:
Why is a 16bit LUT calculator treated differently and why is that not version specific?

Reply:
LAB encodings only changed for 16-bit LUTs from v2 to v4
- 16-bit max values for 100 & 127 were `FF00` in v2 and became `FFFF` in v4
- 8-bit max values for 100 & 127 were `FF` in v2 and stayed `FF` in v4

But for the LUTs, the 16-bit type continues to use the legacy encoding:
> For colour values that are in the PCSLAB colour space on the PCS side of the tag, this tag uses the legacy 16-
bit PCSLAB encoding defined in Tables 42 and 43, not the 16-bit PCSLAB encoding defined in 6.3.4.2. This
encoding is retained for backwards compatibility with profile version 2.